src/middleware/rewriteUrls.coffee

```diff
@@ -0,0 +1,104 @@
+url = require 'url'
+winston = require 'winston'
+Q = require 'q'
+
+Channel = (require '../../lib/model/channels').Channel
+config = require '../config/config'
+routerConf = config.get 'router'
+utils = require '../../lib/utils'
+router = require '../../lib/middleware/router'
+
+invertPathTransform = (pathTransform) ->
+  return pathTransform.replace /s\/(.*?)\/(.*?)($|\/(.*)$)/, 's/$2/$1/$4'
+
+exports.fetchRewriteConfig = (channel, authType, callback) ->
+  # set rewrite config from current channel
+  rwConfig = []
+  if channel.rewriteUrlsConfig?
+    rwConfig = channel.rewriteUrlsConfig
+
+  # add in default rewrite rules for hosts we proxy
+  utils.getAllChannels (err, channels) ->
+    if err?
+      return callback err
+
+    for channel in channels
```

Comment:
Well not really, what it does is finds which hosts we proxy that are on primary routes, then it creates 'virtual' rules for them. So for example if the SHR for some reason sent back a link to a patient in the CR (using a direct link to the CR), then if we have a channel that points to the CR on a primary route we are able to rewrite the link to point to us instead because we know that host. It doesn't add the static, user configurable config from other channel, those are only applied per channel only, but it does create defaults for other  routes that we proxy. Does that make sense, if kinda gets a bit complex?

Perhaps we should have a config option on the channel for this, something like `autoRewriteUrlsToOtherChannels`?

:bulb: Ah, wait, looking at the code what I just explained isn't how it actually works, this is a bug, it shouldn't include user config for all channels, the the current one. You are right. Good catch.